app/uk/gov/hmrc/gform/formtemplate/Rewriter.scala

```diff
@@ -162,9 +162,29 @@ trait Rewriter {
     val includeIfs: List[IncludeIf] = formTemplate.formKind.allSections.flatMap {
       case Section.NonRepeatingPage(page) => page.includeIf.toList
       case Section.RepeatingPage(page, _) => page.includeIf.toList
-      case Section.AddToList(_, _, _, _, _, _, _, includeIf, _, pages, _, _, _, _, _, _, fields, _) =>
+      case Section.AddToList(
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            includeIf,
+            pages,
+            repeatsUntil,
+            repeatsWhile,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            fields,
+            _
+          ) =>
         includeIf.toList ++ pages.toList.flatMap(_.includeIf.toList) ++ fields.fold(List.empty[IncludeIf])(
-          _.toList.flatMap(_.includeIf.toList)
+          _.toList.flatMap(_.includeIf.toList) ++ repeatsUntil.toList ++ repeatsWhile.toList
```

Comment:
This should not assume there must be `fields` field.

Reply:
Fixed